lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -458,13 +465,16 @@ def _download_file(url, b_path, expected_hash, validate_certs, token=None, timeo
     display.display("Downloading %s to %s" % (url, to_text(b_tarball_dir)))
     # NOTE: Galaxy redirects downloads to S3 which rejects the request
     # NOTE: if an Authorization header is attached so don't redirect it
-    resp = open_url(
-        to_native(url, errors='surrogate_or_strict'),
-        validate_certs=validate_certs,
-        headers=None if token is None else token.headers(),
-        unredirected_headers=['Authorization'], http_agent=user_agent(),
-        timeout=timeout
-    )
+    try:
+        resp = open_url(
+            to_native(url, errors='surrogate_or_strict'),
+            validate_certs=validate_certs,
+            headers=None if token is None else token.headers(),
+            unredirected_headers=['Authorization'], http_agent=user_agent(),
+            timeout=timeout
+        )
+    except Exception as err:
+        raise AnsibleError(to_native(err), orig_exc=err)
```

Comment:
It'd be nice to avoid raising a generic `AnsibleError` from the internal layers. I know that there's some leftovers in here but raising more such non-specific exceptions is only making the matter worse. A `RuntimeError`, which I believe his exception conversion represents, should be more than enough. Although, the exceptions that may happen at this stage are probably related to the initial request/connection (since the consumption is happening later). So maybe a [`ConnectionError`](https://docs.python.org/3/library/exceptions.html#ConnectionError) would be more accurate.

Reply:
This exception is raised to the user, so then we'd need to catch a RuntimeError and raise an AnsibleError. I like the current implementation, rather than creating layers and layers of unnecessary exception handling.

Reply:
Isn't it being converted in `api.py` already? Also, only the outer layer should be interacting with the user, otherwise abstractions leak.

Reply:
`api.py` does nothing for conversions. For one it's in `module_utils` and doesn't have access to raise `AnsibleError`.

It simply re-raises whatever exception is encountered. 

Unfortunately, we'd have a lot of work to do, to not have leaky abstractions.  As it is, so much of the code base operates under the assumption that all layers raise `AnsibleError` for direct user feedback, and we handle it in the CLI code.  Otherwise, it's just an arbitrary unhandled exception type.  I actually wasn't doing this originally, and we had a bunch of integration tests that started failing.

An exercise for future consideration.